aroquev/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -60,4 +69,21 @@ public Entity toEntity() {
     return commentEntity;
   }
 
+  /**
+   * Returns the nickname of the user with id, or empty String if the user has not set a nickname.
+   */
+  private String getUserNickname(String id) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query =
+        new Query("UserInfo")
+            .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    if (entity == null) {
```

Comment:
FYI you could return an `Optional<String>` from this method to better communicate that a username could be empty. 

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

```java
if (entity == null) {
  return Optional.empty();
}
// ...
return Optional.of(nickname);
```

```java
Optional<String> nickname = getUserNickname(...);
if (!nickname.isPresent()) {...}
else {
  System.out.println(nickname.get());
}
```

Reply:
Nice, would it be good that the attribute of Comment is also an optional, or that one should remain as a String?

Reply:
You'd probably want to change that to
```java
this.nickname = getUserNickname(username).orElse("anonymous");
```

Reply:
Done.